### PR TITLE
solaar: 1.1.3 -> 1.1.4 and clean up

### PR DESCRIPTION
**Pull request by oxalica**
###### Description of changes

1. Version bump. Also some build inputs are not necessary and are removed.
2. Remove the udev rule wrapper `logitech-udev-rules` by pointing it to an individual output of solaar.

Tested with G102 mouse on none+swaywm.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Built successfully. Tried to run the `solaar` app, but:

```
[nix-shell:~/.cache/nixpkgs-review/pr-180191-1]$ results/solaar/bin/solaar
Gtk-Message: 13:53:29.751: Failed to load module "colorreload-gtk-module"

(.solaar-wrapped:12367): Gtk-WARNING **: 13:53:29.766: Found an icon but could not load it. Most likely gdk-pixbuf does not provide SVG support.
Traceback (most recent call last):
  File "/nix/store/ql2b37hi8hhkgz84dvzr4gmpfh8ika5p-solaar-1.1.4/lib/python3.10/site-packages/solaar/ui/__init__.py", line 169, in <lambda>
    application.connect('startup', lambda app, startup_hook: _startup(app, startup_hook, use_tray, show_window), startup_hook)
  File "/nix/store/ql2b37hi8hhkgz84dvzr4gmpfh8ika5p-solaar-1.1.4/lib/python3.10/site-packages/solaar/ui/__init__.py", line 116, in _startup
    window.init(show_window, use_tray)
  File "/nix/store/ql2b37hi8hhkgz84dvzr4gmpfh8ika5p-solaar-1.1.4/lib/python3.10/site-packages/solaar/ui/window.py", line 829, in init
    Gtk.Window.set_default_icon_from_file(_icons.icon_file(NAME.lower()))
TypeError: Argument 0 does not allow None as a value
Traceback (most recent call last):
  File "/nix/store/ql2b37hi8hhkgz84dvzr4gmpfh8ika5p-solaar-1.1.4/lib/python3.10/site-packages/solaar/ui/__init__.py", line 134, in _command_line
    _activate(app)
  File "/nix/store/ql2b37hi8hhkgz84dvzr4gmpfh8ika5p-solaar-1.1.4/lib/python3.10/site-packages/solaar/ui/__init__.py", line 127, in _activate
    app.add_window(window._window)
TypeError: Argument 1 does not allow None as a value
```

File: pkgs/applications/misc/solaar/default.nix [hunk 1]
```diff
@@ -4,26 +4,27 @@
...
   src = fetchFromGitHub {
     owner = "pwr-Solaar";
     repo = "Solaar";
     rev = version;
-    hash = "sha256-6z22MnhUL9Da3G7UDmZsBAi2gHLNpiFEwe+pAtnP91s=";
+    hash = "sha256-nDfVF7g0M54DRpkH1rnZB8o+nCV4A6b1uKMOMRQ3GbI=";
   };
 
-  nativeBuildInputs = [ wrapGAppsHook gdk-pixbuf ];
-  buildInputs = [ libappindicator librsvg ];
```
Comment (viraptor) [inline pkgs/applications/misc/solaar/default.nix:26]: Both gdk-pixbuf and librsvg are required. I validated that removing either one results in the "Found an icon but could not load it. Most likely gdk-pixbuf does not provide SVG support." error.

**Review by viraptor — CHANGES_REQUESTED** [review]
Confirmed the libraries are required for the gui.

File: pkgs/applications/misc/solaar/default.nix (hunk 1 above)
Comment (oxalica) [reply pkgs/applications/misc/solaar/default.nix]: Oops, the icon works fine in my environment. Maybe because I'm on swaywm?
Reverted now.

**Review by viraptor — APPROVED** [review]
Result of `nixpkgs-review pr 180191` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>solaar</li>
  </ul>
</details>

Don't have any devices to test with, but the app seems to work as expected otherwise. If I understand correctly, the migration should be transparent for people referencing `logitech-udev-rules` too.

The rules seem to be installed in `${solaar}/share/solaar/udev-rules.d/` as well. Do you want to deduplicate/remove them, or just leave it be?

File: pkgs/applications/misc/solaar/default.nix [hunk 2]
```diff
@@ -41,12 +43,9 @@ python3Packages.buildPythonApplication rec {
   postInstall = ''
     ln -s $out/bin/solaar $out/bin/solaar-cli
 
-    install -Dm444 -t $out/etc/udev/rules.d rules.d/*.rules
+    install -Dm444 -t $udev/etc/udev/rules.d rules.d-uinput/*.rules
```
Comment (bjornfor) [inline pkgs/applications/misc/solaar/default.nix:47]: `rules.d-uinput/`? Either a mistake or I need to learn what that is.

Comment (viraptor) [reply pkgs/applications/misc/solaar/default.nix:47]: It's a path in the repo: https://github.com/pwr-Solaar/Solaar/tree/master/rules.d-uinput

**Review by bjornfor — APPROVED** [review]

File: pkgs/applications/misc/solaar/default.nix (hunk 2 above)
Comment (oxalica) [reply pkgs/applications/misc/solaar/default.nix:47]: According to https://github.com/pwr-Solaar/Solaar/commit/a40b9efdcaf2c2362d46b7237b4d0b2baeb8c45d, the new udev rules contains an additional rule of `/dev/uinput`, which would hopefully fix an permission issue under wayland.

**Comment by oxalica** [comment]
> The rules seem to be installed in `${solaar}/share/solaar/udev-rules.d/` as well. Do you want to deduplicate/remove them, or just leave it be?

The file is small for users who are already installing `solaar.out`. I'd keep it as is.
I think the split of udev rules is mainly for users who only wants the udev rules without pulling python libraries, but not the reverse.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -24846,7 +24846,9 @@ with pkgs;
 
   logiops = callPackage ../misc/drivers/logiops { };
 
-  logitech-udev-rules = callPackage ../os-specific/linux/logitech-udev-rules { };
+  # ltunifi and solaar both provide udev rules but solaar's rules are more
+  # up-to-date so we simply use that instead of having to maintain our own rules
+  logitech-udev-rules = solaar.udev;
```
Comment (lovesegfault) [inline pkgs/top-level/all-packages.nix:24851]: This should go into `pkgs/top-level/aliases.nix`

Comment (bjornfor) [reply pkgs/top-level/all-packages.nix:24851]: > This should go into `pkgs/top-level/aliases.nix`

...only if `logitech-udev-rules` is a backwards compatibility alias, right? I interpret it as a persistent alias, not something that should be removed, so I think the current location is fine. I guess @oxalica can clarify.

Comment (oxalica) [reply pkgs/top-level/all-packages.nix:24851]: > > This should go into `pkgs/top-level/aliases.nix`
> 
> ...only if `logitech-udev-rules` is a backwards compatibility alias, right? I interpret it as a persistent alias, not something that should be removed, so I think the current location is fine. I guess @oxalica can clarify.

+1. I think `aliases.nix` is for deprecated names or persistent renaming. Just like why we not putting `rust = rust_1_61;
` in it.

Comment (lovesegfault) [reply pkgs/top-level/all-packages.nix:24851]: I don't care about this point enough, happy with whatever you choose.

**Pull request merged**